### ISSUE TITLE
G09 (P2) -- EMB : le pooling final inclut les positions de padding

### ISSUE THREAD

**Issue by jsboige** [jsboige/CoursIA#15045]
Fille de #15035 (audit Astra #3, 2026-09-07). Le cadre de lecture, la methode, la verification de fraicheur et **la correction du user sur le protocole des TP** sont dans l'EPIC -- les lire avant d'agir.

**Etiquetage d'Astra conserve tel quel** : **P2 avant d'utiliser le pont vers le transformer comme reference. VERIFIE source et arithmetique ; pas d'inference Transformer executee.**

## Le defaut

Le tokenizer recoit `padding=True` et le modele recoit les tenseurs ; l'embedding est ensuite `o.last_hidden_state.mean(dim=1)`, **sans appliquer `attention_mask` a la moyenne**. `[EMB:c19]`

Meme si l'attention du modele utilise son masque, **la moyenne finale** inclut toujours les lignes de padding. Le nombre de lignes moyennees depend de **la phrase la plus longue du lot**. Une meme phrase peut donc **changer de representation selon ses voisines de batch**.

Illustration arithmetique du defaut -- **pas une sortie de ce modele** :

```text
deux positions utiles [[1,0],[1,0]]          -> moyenne [1.0, 0.0]
avec deux pads [[0,2],[0,2]] non neutralises -> moyenne [0.5, 1.0]
```

## Correction (Astra)

Moyenne **ponderee par `attention_mask`**, division par le **nombre de positions utiles** ; preciser la normalisation utilisee et le lien avec le modele de sentence embeddings choisi.

## Acceptation (verbatim)

> L'embedding d'une phrase seule et celui de cette meme phrase dans un lot contenant une phrase beaucoup plus longue **coincident a la tolerance numerique choisie, en mode evaluation**. **Ce test doit etre fait sur le vrai modele avant de clore le finding.**

La derniere phrase est une contrainte de cloture, pas un detail : Astra n'a **pas** execute d'inference Transformer. Fermer ce finding sur la seule lecture du source reproduirait le defaut qu'il decrit -- un raisonnement arithmetique n'est pas une mesure.

**Comment by jsboige** [comment]
[CLAIMED] G09 (pooling attention_mask) — myia-po-2023:CoursIA — 2026-09-07T14:52+02:00. Fix cellule 19 (moyenne pondérée par attention_mask) + test de coïncidence solo/lot sur le vrai modèle en mode évaluation, avant clôture.
